test/document-util.test.js

```diff
@@ -159,12 +159,13 @@ describe('Document utility tests', () => {
                         expect(!!imposter).toStrictEqual(!!hasImposter);
 
                         const range = document.createRange();
-                        range.setStart(/** @type {Node} */ (imposter ? imposter : startNode), startOffset);
-                        range.setEnd(/** @type {Node} */ (imposter ? imposter : startNode), endOffset);
+                        range.setStart(/** @type {Node} */ (imposter ?? startNode), startOffset);
+                        range.setEnd(/** @type {Node} */ (imposter ?? startNode), endOffset);
 
                         // Override getClientRects to return a rect guaranteed to contain (x, y)
                         range.getClientRects = () => {
                             /** @type {import('test/document-types').PseudoDOMRectList} */
+                            // eslint-disable-next-line sonarjs/prefer-immediate-return
```

Comment:
There are a few cases where I had to disable `sonarjs/prefer-immediate-return` because it needs the type hint. I could have used a cast, but a type hint is a bit safer and the syntax a bit cleaner.